2024-01-24 13:43:09: Experiment log path in: D:\omer\ST-SSL\experiments\NYCBike1\20240124-134309
2024-01-24 13:43:09: Experiment configs are: Namespace(batch_size=32, best_path='None', d_input=2, d_model=64, d_output=2, data_dir='data', dataset='NYCBike1', debug=False, device='cuda', dropout=0.2, early_stop=True, early_stop_patience=15, epochs=100, grad_norm=True, graph_file='data/NYCBike1/adj_mx.npz', input_length=19, log_dir='D:\\omer\\ST-SSL\\experiments\\NYCBike1\\20240124-134309', lr_init=0.001, max_grad_norm=5, mode='train', nmb_prototype=6, num_nodes=128, percent=0.1, seed=31, shm_temp=0.5, temp=4, test_batch_size=32, use_dwa=True, yita=0.6)
2024-01-24 13:43:09: Traceback (most recent call last):
  File "main.py", line 59, in model_supervisor
    results = trainer.train() # best_eval_loss, best_epoch
  File "D:\omer\ST-SSL\model\trainer.py", line 109, in train
    train_epoch_loss, loss_t = self.train_epoch(epoch, loss_weights)
  File "D:\omer\ST-SSL\model\trainer.py", line 57, in train_epoch
    repr1, repr2 = self.model(data, self.graph) # nvc
  File "C:\Users\IST\.conda\envs\ST-SSL\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\IST\.conda\envs\ST-SSL\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\omer\ST-SSL\model\models.py", line 32, in forward
    repr1 = self.encoder(view1, graph) # view1: n,l,v,c; graph: v,v
  File "C:\Users\IST\.conda\envs\ST-SSL\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\IST\.conda\envs\ST-SSL\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\omer\ST-SSL\model\layers.py", line 207, in forward
    x = x.permute(0, 3, 1, 2)  # (batch_size, feature_dim, input_length, num_nodes), nclv
RuntimeError: permute(sparse_coo): number of dimensions in the tensor input does not match the length of the desired ordering of dimensions i.e. input.dim() = 3 is not equal to len(dims) = 4

